Deploy Apache Spark on Google Axion C4A VM (#9)
odidev/arm-learning-paths: content/learning-paths/servers-and-cloud-computing/spark-on-gcp/_index.md

---
title: Deploy Apache Spark on Google Axion C4A Virtual machine

minutes_to_complete: 60

> who_is_this_for: This is an advanced topic for anyone who wants to deploy Spark on GCP C4A Arm64 virtual machine, including installation, baseline testing, and benchmarking. 

Reviewer: Kindly keep this syntax same for all LPs unless pinged by Arm. 
```suggestion
who_is_this_for: This is an introductory topic for the software developers who are willing to migrate their Apache Spark workloads from the x86_64 platforms to Arm-based platforms, or on Google Axion-based C4A virtual machines specifically. 
```

learning_objectives:

>        - Provision an Arm virtual machine on the Google Cloud Platform using the C4A Google Axion instance family.

Reviewer: ```suggestion
       - Provision an Arm virtual machine on the Google Cloud Platform using the C4A Google Axion instance family, and RHEL 9 as the base image.
```

       - Understand how to install and configure Apache Spark on Arm-based GCP C4A instances.
       - Deploy Spark in local mode and validate functionality through baseline testing.

>        - Perform benchmarking to evaluate and compare Apache Spark’s performance on both x86_64 and Arm.

Reviewer: ```suggestion
       - Perform benchmarking to evaluate and compare Apache Spark’s performance on Arm.
```
